jetson/src/core/robot.cpp

```diff
@@ -73,34 +73,32 @@ void Robot::updateArmPosition() {
 	armPosition[2] = servoArm[2].getElapsedDistance();
 }
 
-// From MTE 544 Control Lecture Slide 9
-double calculate_radius(double delta_x, double delta_y) {
-	double L = sqrt(delta_x * delta_x + delta_y * delta_y);
-
-	return L * L / 2 * std::abs(delta_y);
-}
-
 // From MTE 544 Modeling III IV Lecture Slide 16
 // Pass in -w for left wheel
 // Assume positive w is CCW orientation
 double Robot::calculate_wheel_speed(double v, double w) {
 	return 1 / WHEEL_RADIUS * (v + CARTESIAN_X_MAX * w / 2 + std::pow(CARTESIAN_Y_MAX * w / 2, 2) / (v + CARTESIAN_X_MAX * w / 2));
 }
 
-void Robot::driveRobotForward(Point2D delta) {
+void Robot::driveRobotForward(Point2D goal) {
+	double angular_error_gain = 2;
 	robotPosition = Point2D(0, 0);
 	drive[frontLeft].resetElapsedDistance();
 	drive[backLeft].resetElapsedDistance();
 	drive[frontRight].resetElapsedDistance();
 	drive[backRight].resetElapsedDistance();
+	Point2D delta = goal - robotPosition;
 
 	while (delta.mag() < ROBOT_TOL) {
```

Comment:
Not sure what suggestion to make but I'd like to just acknowledge that an overshoot would result in a runaway robot. Or at least a spiraling robot